api/server/server.go

```diff
@@ -42,9 +50,5 @@ func (tapi ToDDApi) Start(cfg config.Config) {
 	serve_url := fmt.Sprintf("%s:%s", tapi.cfg.API.Host, tapi.cfg.API.Port)
 
 	log.Infof("Serving ToDD Server API at: %s\n", serve_url)
-	err := http.ListenAndServe(serve_url, nil)
-	if err != nil {
-		log.Error("Error starting API")
-		os.Exit(1)
-	}
+	return http.ListenAndServe(serve_url, nil)
```

Comment:
The most common type of error from `http.ListenAndServer` is returned immediately if it can't bind to the IP/port given. Do you have an example of what you would like to see?